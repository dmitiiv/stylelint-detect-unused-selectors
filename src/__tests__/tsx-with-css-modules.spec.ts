import path from 'path';
import stylelint from 'stylelint';
import { parseResult } from './__helpers__/parse-result';
import { getLintOptions } from './__helpers__/get-lint-option';

const fixturesRoot = path.join(
  __dirname,
  '..',
  '..',
  'examples',
  'tsx-with-css-modules',
);

test('Disallow selectors with warning that are not used in a TSX file using CSS Modules', async (): Promise<
  void
> => {
  const result = await stylelint.lint(getLintOptions('warning', fixturesRoot));

  expect(result.results[0].warnings).toHaveLength(1);
  expect(parseResult(result)).toMatchSnapshot();
});

test('Disallow selectors with error that are not used in a TSX file using CSS Modules', async (): Promise<
  void
> => {
  const result = await stylelint.lint(getLintOptions('error', fixturesRoot));

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});
