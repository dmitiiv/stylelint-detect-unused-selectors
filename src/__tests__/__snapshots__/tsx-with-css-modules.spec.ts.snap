// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabllow selectors that are not used in a TSX file using CSS Modules 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 22,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.tsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;
